amqp/consumer.go

```diff
@@ -247,33 +355,59 @@ func (c *Consumer) Consume() {
 		)
 
 		if err != nil {
-			log.WithFields(log.Fields{
-				"type":    "goevents",
-				"subType": "consumer",
-				"queue":   c.queueName,
-				"error":   err,
+			logger.WithFields(log.Fields{
+				"queue": c.queueName,
+				"error": err,
 			}).Error("Error setting up consumer...")
 
 			time.Sleep(c.config.consumeRetryInterval)
 
 			continue
 		}
 
-		log.WithFields(log.Fields{
-			"type":    "goevents",
-			"subType": "consumer",
-			"queue":   c.queueName,
+		logger.WithFields(log.Fields{
+			"queue": c.queueName,
 		}).Info("Consuming messages...")
 
 		for m := range msgs {
-			c.dispatch(m)
+			if len(m.MessageId) == 0 {
+				m.MessageId, _ = NewUUIDv4()
```

Comment:
Wondering why messageId is being generated on the consumer's side? Why not in the producer's side?